File: data/EXP01/golang_no_system_var/2420/2420.output_3.go
func goodIndices(nums []int, k int) []int {
    n := len(nums)

    var result []int

    for i := k; i < n-k; i++ {
        good := true

        for j := i-k; j < i; j++ {
            if nums[j] < nums[j+1] {
                good = false
                break
            }
        }

        for j := i; j < i+k; j++ {
            if nums[j] > nums[j+1] {
                good = false
                break
            }
        }

        if good {
            result = append(result, i)
        }
    }

    return result
}

File: data/EXP01/python/2420/2420.output_3.py
# @lc app=leetcode id=2420 lang=python3
from typing import List

class Solution:
    def goodIndices(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        result = []
        
        for i in range(k, n - k):
            if self.isGoodIndex(nums, i, k):
                result.append(i)
                
        return result
    
    def isGoodIndex(self, nums: List[int], idx: int, k: int) -> bool:
        for i in range(k):
            if nums[idx - i] < nums[idx - i - 1] or nums[idx + i] > nums[idx + i + 1]:
                return False
        return True